### PR TITLE
7 write some tests

### LINKED ISSUE #7: Write some tests
Test everything that is implemented now.

### PR DESCRIPTION
**Pull request by ebolloff**
This should fix #7.
It fixes some code, adds bunnymock support for X-Recent-History-Exchange and tests all of the code.

### CODE REVIEW

File: spec/support/bunnymock_recent_history_exchange.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+require 'bunny-mock'
+
+module BunnyMock
+  class Exchange
+    # Monkeypatch, XRecentHistory be found by the ex declare "x-recent-history"
+    def self.declare(channel, name = '', opts = {})
```
Comment (ebolloff) [inline spec/support/bunnymock_recent_history_exchange.rb]: I can't change that because its a method from `bunny-mock`.

Comment (eugenk) [inline spec/support/bunnymock_recent_history_exchange.rb]: You could write `def self.declare(channel, name = '', **opts)` which keeps the interface and is okay with rubocop.

File: Gemfile.lock [hunk 2]
```diff
@@ -78,19 +84,24 @@ GEM
...
 DEPENDENCIES
-  bundler (~> 1.11)
+  awesome_print (~> 1.7.0)
+  bundler (~> 1.12)
+  bunny-mock (~> 1.5.0)
   coveralls (~> 0.8.17)
   hets-rabbitmq-wrapper!
+  pry (~> 0.10.4)
   pry-byebug (~> 3.4.2)
+  pry-coolline (~> 0.2.5)
```
Comment (eugenk) [inline Gemfile.lock]: I thought, I removed this one... See #21

Comment (eugenk) [inline Gemfile.lock]: My bad. I didn't see it in the gemspec.

File: hets-rabbitmq-wrapper.gemspec [hunk 3]
```diff
@@ -26,9 +26,14 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.add_development_dependency 'bundler', '~> 1.11'
+  spec.add_development_dependency 'bundler', '~> 1.12'
   spec.add_development_dependency 'rake', '~> 12.0.0'
   spec.add_development_dependency 'rspec', '~> 3.5.0'
+  spec.add_development_dependency 'pry', '~> 0.10.4'
+  spec.add_development_dependency 'pry-byebug', '~> 3.4.0'
+  spec.add_development_dependency 'pry-coolline', '~> 0.2.5'
```
Comment (eugenk) [inline hets-rabbitmq-wrapper.gemspec]: Please don't include pry-coolline. See https://github.com/ontohub/ontohub-models/pull/24

File: spec/hets-rabbitmq-wrapper_spec.rb [hunk 4]
```diff
@@ -1,9 +1,39 @@
...
+  context 'minimal parsing version queue' do
+    it 'subscribe to' do
+      expect(session.queue_exists?('q_min_parsing_version')).to be_truthy
+    end
+  end
+
+  context 'worker queue' do
+    before do
+      allow(subscriber).
+        to receive(:call_hets_version).and_return('v0.99, 1471209385')
```
Comment (eugenk) [inline spec/hets-rabbitmq-wrapper_spec.rb]: Please put the string into a variable for re-use, like
```ruby
let(:version_timestamp) { 1471209385 }
let(:version) { "v0.99, #{version_timestamp}" }

before do
  ...
```

File: spec/hets-rabbitmq-wrapper_spec.rb [hunk 5]
```diff
@@ -1,9 +1,39 @@
...
+      queue.publish('v0.99, 1471209385')
+    end
+
+    it 'subscribe to' do
```
Comment (eugenk) [inline spec/hets-rabbitmq-wrapper_spec.rb]: How about `'subscribes successfully'`?

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -1,6 +1,14 @@
...
+require_relative 'support/simplecov'
+
 $LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
 
 require 'hets-rabbitmq-wrapper'
-require_relative 'support/simplecov'
+require 'bunny-mock'
+
+RSpec.configure do |config|
+  config.before(:each) do
```
Comment (eugenk) [inline spec/spec_helper.rb]: Is it necessary to do this before `:each` (repeatedly before every single example) or is it okay to do it before `:suite` (before the first example only)?

File: spec/subscriber_spec.rb [hunk 7]
```diff
@@ -14,7 +14,7 @@
     end
 
     it 'parses the version correctly' do
-      expect(subject.hets_version).to eq(1471209385)
+      expect(subject.hets_version).to eq(1_471_209_385)
```
Comment (eugenk) [inline spec/subscriber_spec.rb]: Please put it in a variable as asked for in spec/hets-rabbitmq-wrapper_spec.rb.

File: spec/support/bunnymock_recent_history_exchange.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+require 'bunny-mock'
+
+module BunnyMock
+  class Exchange
+    # Monkeypatch, XRecentHistory be found by the ex declare "x-recent-history"
+    def self.declare(channel, name = '', **opts)
+      # get requested type
+      type = opts.fetch :type, :direct
+      # get needed class type
+      klazz = BunnyMock::Exchanges.const_get(type.to_s.split('-').
```
Comment (eugenk) [inline spec/support/bunnymock_recent_history_exchange.rb]: try `type.to_s.underscore.camelize` ;)

Comment (ebolloff) [inline spec/support/bunnymock_recent_history_exchange.rb]: `underscore` works in Rails, but not in the gem, so I don't change it.

**Pull request merged**